### PR TITLE
Add create, delete and edit card/row controls

### PR DESCRIPTION
**Pull request by connoratrug**
- [x] Add add row/card button to toolbar
   - Add button with link to data-row-edit plugin
   - Do not show button in cart view
- [x] Edit row / card
- [x] Delete row(s)/ card

#### Checklist
- [x] Functionality works & meets specifications
- [x] Code reviewed
- [x] Code unit/integration/system tested
- [x] User documentation updated
- [x] Conventional commits (squash if needed)
- [x] No warnings during install
- [x] Updated javascript typing

### CODE REVIEW

File: packages/data-explorer/src/components/dataView/DefaultCardContent.vue [hunk 1]
```diff
@@ -1,6 +1,19 @@
 <template>
   <div>
     <h5 class="card-title mg-default-card-title">{{dataLabel}}
+      <a
+        class="btn btn-sm btn-link ml-1"
+        role="button"
+        :href="'/plugin/data-row-edit/' + dataTable + '/' + dataId">
```
Comment (MaxPostema) [inline packages/data-explorer/src/components/dataView/DefaultCardContent.vue:7]: Add "plugin" to devServer proxy to be able to test the changes locally

File: packages/data-explorer/src/components/dataView/TableHeader.vue [hunk 2]
```diff
@@ -2,6 +2,7 @@
   <thead>
   <tr>
     <th scope="col" v-if="isShop"></th>
+    <th scope="col" v-if="!isShop"></th>
```
Comment (MaxPostema) [inline packages/data-explorer/src/components/dataView/TableHeader.vue:5]: I don't see the reason to do this.. Is this useful?

File: packages/data-explorer/src/components/dataView/TableRow.vue [hunk 3]
```diff
@@ -1,19 +1,45 @@
 <template>
   <tr>
     <td v-if="isShop"><shopping-button :isSelected="isSelected" :id="id"></shopping-button></td>
+    <td v-if="!isShop">
```
Comment (MaxPostema) [inline packages/data-explorer/src/components/dataView/TableRow.vue:4]: In the default card view i can do shopping AND crud operations. This makes the table only have one or the other. 
I argue to make in consistent. 
Especially if we (later) add options to for example mass remove items via "shopping" selections. ( then you would like to be able to add to card in tableview. )

File: packages/data-explorer/src/components/dataView/TableRow.vue [hunk 4]
```diff
@@ -1,19 +1,45 @@
...
+      <div class="btn-group" role="group" aria-label="row actions">
+        <button
+          class="btn btn-sm btn-link"
+          role="button"
+          @click="$eventBus.$emit('delete-item', id)"
```
Comment (MaxPostema) [inline packages/data-explorer/src/components/dataView/TableRow.vue:9]: Nice trick. :)

File: packages/data-explorer/src/repository/dataRepository.ts [hunk 5]
```diff
@@ -140,8 +140,13 @@ const getRowDataWithReferenceLabels = async (tableId: string, rowId: string, met
   return levelOneRowMapper(response.data, metaData)
 }
 
+const deleteRow = async (tableId: string, rowId: string) => {
+  return axios.delete(`/api/data/${tableId}/${rowId}`)
```
Comment (MaxPostema) [inline packages/data-explorer/src/repository/dataRepository.ts:144]: Should there be state handeling if delete fails?... maybe the user is not allowed to delete this row and should be notified?

Comment (connoratrug) [reply packages/data-explorer/src/repository/dataRepository.ts:144]: yes , should be passed to the action and handled there imo

File: packages/data-explorer/src/components/dataView/TableRow.vue (hunk 4 above)
Comment (connoratrug) [reply packages/data-explorer/src/components/dataView/TableRow.vue:9]: tnks, only one of the possible options, suggest to discuss these during the next meeting about the design

File: packages/data-explorer/src/components/dataView/TableRow.vue (hunk 3 above)
Comment (connoratrug) [reply packages/data-explorer/src/components/dataView/TableRow.vue:4]: i'm sorry i don't understand what you are saying here, do you suggest to add crud operations to the cart view ?, i would expect there to be replaced with cart actions ( remove from cart )

File: packages/data-explorer/src/components/dataView/TableHeader.vue (hunk 2 above)
Comment (connoratrug) [reply packages/data-explorer/src/components/dataView/TableHeader.vue:5]: the column is there for two different reasons ( shop stuff or actions stuff), i agree the code looks silly now , i wanted to make it explicit so we can refactor this once we are clear on the UI/UX design

File: packages/data-explorer/src/components/dataView/DefaultCardContent.vue (hunk 1 above)
Comment (connoratrug) [reply packages/data-explorer/src/components/dataView/DefaultCardContent.vue:7]: o cool does work ? i will try it , good suggestion

**Comment by connoratrug** [comment]
> There are some feedback items and a merge conflict.

merged with master and resolved conflict 

> ( ps: It was a bit annoying to have my state disappear after every crud operation. I hope that storing state in URL will improve the flow.)

Yes it will, however add and edit actions as separate app where designed to combine react bs-3 explorer with vue bs-4 app, to improve ux we could/should move add and edit actions into same app ( separate vue of modal) and use vue routing to switch view instead of page load.

**Pull request merged**